src/lists.rs

```diff
@@ -28,6 +28,10 @@ pub enum ListTactic {
     Horizontal,
     // Try Horizontal layout, if that fails then vertical.
     HorizontalVertical,
+    // Try Horizontal layout, if that fails then put first element on the first line,
+    // and others as many items as possible per row over (possibly) many rows.
+    // Used for macro calls like `format!`, `println!`, etc
+    FormatLikeMacroCall(usize),
```

Comment:
I don't think it is quite right to add this variant to this enum - one could imagine combining any of these formatting styles with the format-macro style of putting the string lit on its own line. I.e., this should probably be a separate flag into the list formatting functions.